btrack/utils.py

```diff
@@ -230,6 +235,75 @@ def nans_idx(x):
     return data, properties, graph
 
 
+def napari_to_tracks(
+    data: npt.NDArray,
+    properties: Optional[dict[str, npt.ArrayLike]],
+    graph: Optional[dict[int, list[int]]],
+) -> list[btypes.Tracklet]:
+    """Convert napari Tracks to a list of Tracklets.
+
+    Parameters
+    ----------
+    data : array (N, D+1)
+        Coordinates for N points in D+1 dimensions. ID,T,(Z),Y,X. The first
+        axis is the integer ID of the track. D is either 3 or 4 for planar
+        or volumetric timeseries respectively.
+    properties : dict {str: array (N,)}
+        Properties for each point. Each property should be an array of length N,
+        where N is the number of points.
+    graph : dict {int: list}
+        Graph representing associations between tracks. Dictionary defines the
+        mapping between a track ID and the parents of the track. This can be
+        one (the track has one parent, and the parent has >=1 child) in the
+        case of track splitting, or more than one (the track has multiple
+        parents, but only one child) in the case of track merging.
+
+    Returns
+    -------
+    tracks : list[btypes.Tracklet]
+        A list of tracklet objects from BayesianTracker.
+
+    """
+
+    if data.shape[1] == Dimensionality.FIVE:
+        track_id, t, z, y, x = data.T
+    elif data.shape[1] == Dimensionality.FOUR:
+        track_id, t, y, x = data.T
+        z = np.zeros_like(x)
+    else:
+        raise ValueError(
+            "Data must have either 4 (ID, t, y, x) or 5 (ID, t, z, y, x) columns, "
+            f"not {data.shape[1]}"
+        )
+
+    # Create all PyTrackObjects
+    objects_dict = {
+        "ID": track_id,
+        "t": t,
+        "x": x,
+        "y": y,
+        "z": z,
+        "dummy": properties["dummy"],
+        "label": properties["state"],
+    }
+    track_objects = objects_from_dict(objects_dict)
+
+    # Create all Tracklets
+    tracklets = []
+    for track in np.unique(track_id).astype(int):
+        track_indices = np.argwhere(track_id == track).ravel()
+        track_data = [track_objects[i] for i in track_indices]
+        tracklet = Tracklet(
+            ID=track,
+            data=track_data,
+        )
+        tracklet.root = properties["root"][track_indices][0]
+        tracklet.parent = properties["parent"][track_indices][0]
```

Comment:
I think this'll work most of the time, except when trying to save a tracks layer that wasn't created with `btrack`, i.e. it doesn't have these properties. It'd be safer to reconstruct `parent`, `root` and `children` from `graph` if it is present in the layer.

Reply:
good point! I'm now using the `graph` to construct this data, but one thing I'm not sure of is how to define `root` if there are multiple `parent`s? As I think root is always a single int?

Reply:
Yes, that's correct. In most cases (originating from btrack) we'll never have more than one parent, so it's safe to assume that that is the root for now

Reply:
okay perfect, thanks